slack_sdk/web/internal_utils.py

```diff
@@ -299,3 +304,124 @@ def _remove_none_values(d: dict) -> dict:
     # '{"a": null, "b": 123}'
     #
     return {k: v for k, v in d.items() if v is not None}
+
+
+def _to_v2_upload_file_item(upload_file: Dict[str, Any]) -> Dict[str, Optional[Any]]:
+    file = upload_file.get("file")
+    content = upload_file.get("content")
+    data: Optional[bytes] = None
+    if file is not None:
+        if isinstance(file, str):  # filepath
+            with open(file.encode("utf-8", "ignore"), "rb") as readable:
+                data = readable.read()
+        else:
+            data = file
+    elif content is not None:
+        if isinstance(content, str):  # filepath
+            data = content.encode("utf-8")
+        elif isinstance(content, bytes):
+            data = content
+        else:
+            raise SlackRequestError("The given content must be either filepath as str or data as bytes")
+
+    filename = upload_file.get("filename")
+    if upload_file.get("filename") is None and isinstance(file, str):
+        # use the local filename if filename is missing
+        if upload_file.get("filename") is None:
+            filename = file.split(os.path.sep)[-1]
+        else:
+            filename = "Uploaded file"
+
+    title = upload_file.get("title", "Uploaded file")
+    if data is None:
+        raise SlackRequestError(f"File content not found for filename: {filename}, title: {title}")
+
+    return {
```

Comment:
Should this method return the prepared file's `alt_text` and `snippet_type` as well? These are two optional fields accepted in the getUploadURLExternal method which appears to be on a file by file basis.

Reply:
Ah, yes! This is a silly bug. Thanks for catching this

Reply:
@srajiang This is fixed ✅